metadb/server.go

```diff
@@ -143,30 +144,36 @@ func (s *service) initStore(ctx context.Context, raftTn *tcp.Layer) (*store.Stor
 
 	// execute any requested join operation
 	if len(joinIPAddresses) > 0 {
-		log.WithContext(ctx).Infof("join addresses are: %v", joinIPAddresses)
-
-		raftAddr := fmt.Sprintf("%s:%d", s.config.ListenAddress, s.config.RaftPort)
-		// join rqlite cluster
-		joinAddr, err := cluster.Join(
-			ctx,
-			"",
-			joinIPAddresses,
-			db.ID(),
-			raftAddr,
-			!s.config.NoneVoter,
-			defaultJoinAttempts,
-			defaultJoinInterval,
-			nil,
-		)
-		if err != nil {
-			return nil, errors.Errorf("join cluster at %v: %w", joinIPAddresses, err)
-		}
-		log.WithContext(ctx).Infof("successfully joined cluster at %v", joinAddr)
+		go s.initClusterJoin(ctx, joinIPAddresses, defaultJoinClusterRetryInterval)
 	}
 
 	return db, nil
 }
 
+func (s *service) initClusterJoin(ctx context.Context, joinAddrs []string, interval time.Duration) {
+	log.WithContext(ctx).Infof("cluster-join initiated - joining in %v", interval)
+	raftAddr := fmt.Sprintf("%s:%d", s.config.ListenAddress, s.config.RaftPort)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
```

Comment:
could you add a log here to know that this init code is existed due to ctx done?